extension/demo/src/modules/storage.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+export async function set (data) {
+  console.log(data)
+  await chrome.storage.local.set(data, console.log)
+
+  return !chrome.runtime.lastError
+}
+
+export async function get (key) {
+  return new Promise(resolve => {
+    chrome.storage.local.get([key], results => {
+      console.log(results[key])
+      resolve(results[key])
+    })
+  })
+}
```

Comment:
A Promise within an `async` function is generally bad practice. This can be rewritten (and simplified) as:

```
export function get (key) {
  return chrome.storage.local.get([key])
}
```

You're returning a promise either way (`async` always makes the return of a function a promise) so there's no need to complicate it.